### PR TITLE
Update cake core

### PR DESCRIPTION
**Pull request by willsmith9182**
upgrade all the things, bump to net472. 
also update pack to not pass output as it is now default parameter

### CODE REVIEW

**Comment by gep13** [comment]
@willsmith9182 thank you very much for taking the time to submit this PR, I really appreciate it!  Updating this addin is something that I have had on my list for a while, but in all honesty, I am not very familiar with how Paket works, and I know @larzw doesn't have as much time to update it as he did previously.  I have reached out to him to ask for a pairing session to go through how to update it, but I haven't had a response.

With regard to the changes that you have made here, can I make a suggestion...

As per the guidelines here:

https://github.com/cake-contrib/Home/blob/master/Guidelines.md#framework

It is actually recommended to compile the addin against .Net 461, as this will mean that the addin has best support across consumers of Cake.  In addition, the ideal would be that it is compiled for netstandard2.0.  The easiest way to achieve this is to convert to the new csproj format, and build with .net core.  Would this change be something that you might be interested in helping with?

**Comment by willsmith9182** [comment]
Yea totally, 

So netstandard2.0 with the new project format, I'll add something to this PR. 

In the newer versions of paket they've changes some command line options, I shall do them on a separate PR after upgrading  to netstandard.

**Comment by gep13** [comment]
@willsmith9182 that is great news!  Thank you for your help, I really appreciate it!

**Comment by marcinsmialek** [comment]
I started working on this .NET Standard migration a few weeks ago, but I stopped as it turned to be way more difficult than I expected because of significant differences in tooling - build process, testing, code coverage checks etc.
I'll try again and if I'm lucky this time, I'll submit the PR, and if not, at least I'd share the partial work, so perhaps it'll be helpful.

**Comment by gep13** [comment]
@marcinsmialek thank you for your offer of help.  To prevent any duplication of effort, can I ask that you and @willsmith9182 work together?  Perhaps best to work on the same branch, or submit PR's into the existing PR branch, that way it will limit the amount of duplication.

**Comment by willsmith9182** [comment]
Yea this seemed "easy" from the outside. 

I've pushed `feature/updateCakeCore` - let's use that as a staging branch. Raise PRs or push to it i'm not fussed, let's see where we get :)

**Comment by willsmith9182** [comment]
image

*kicks various furniture*

netstandard seems to behave fine...

**Comment by gep13** [comment]
@willsmith9182 not sure I follow what is going on here.  Why did you close this PR?  Are you going to open another one?

**Comment by willsmith9182** [comment]
say whaaat. I force pushed back to the beginning (threw away the changes). 

Looks like github decided that means it is closed! 😢

**Comment by gep13** [comment]
Oh, that is weird.  Maybe when you make a new commit it will open again, if not, we can continue discussion on another PR.

**Comment by willsmith9182** [comment]
Yea I'll push something and we'll see what happens. If i make another pr i'll copy all these comments over so we have the info.

**Comment by marcinsmialek** [comment]
Here's my attempt at that upgrade - pretty hefty
https://github.com/willsmith9182/Cake.Paket/pull/1

I haven't touched new command line options so far, as I wanted to have them in another commit. This one feels too big already.

**Comment by willsmith9182** [comment]
HUZZAR! :)

**Comment by willsmith9182** [comment]
from https://github.com/willsmith9182/Cake.Paket/pull/1:

> @marcinsmialek as I have mentioned previously, I have never used Paket before, so I can't speak to how that has been changed, but the other changes look good to me. I will leave the Paket specific changes to yourself and @willsmith9182.
> 
> One thing that I would ask...
> 
> After speaking to the NuGet Team about a few things, the recommendation is to use the following attributes in the the project file:
> 
> <PackageLicenseExpression>MIT</PackageLicenseExpression>
> <RepositoryType>git</RepositoryType>
> <RepositoryUrl>https://github.com/cake-contrib/Cake.Paket.git</RepositoryUrl>
> That way, these attributes flow through and are displayed on the NuGet.org package page.
> 
> Would you be able to add these?

**Comment by willsmith9182** [comment]
@marcinsmialek can you do the project changes and we'll divy up the command changes for the latest paket.exe params

**Comment by willsmith9182** [comment]
Travis is also failing with the following:

```
Installing tools...
Could not find any relevant files for tool 'gitreleasemanager'. Perhaps you need an include parameter?
Error: Failed to install tool 'gitreleasemanager'.
The command "./build.sh" exited with 1.
```

**Comment by marcinsmialek** [comment]
Yeah, I'll do some testing on Ubuntu for Windows, to make sure that build.sh also works.
To be honest, I only tried building on Windows with the PowerShell script.

Later, I'll try to add the rest of required changes - packaging and the fixes related to obsoleted Paket commands. I was too busy to handle it earlier, but there should be plenty of time tomorrow.

**Comment by willsmith9182** [comment]
@marcinsmialek I'll start on the command updates then and leave build stuff to you!

**Comment by willsmith9182** [comment]
@marcinsmialek https://github.com/willsmith9182/Cake.Paket/pull/2 is the command updates
there's also some missing parameters from looking at help sections: https://github.com/cake-contrib/Cake.Paket/issues/30

**Comment by willsmith9182** [comment]
@marcinsmialek I'll have on the drop in code coverage :)

**Comment by marcinsmialek** [comment]
Previously I didn't change paket restore to dotnet restore:
https://fsprojects.github.io/Paket/paket-and-dotnet-cli.html

..and now it appears, that dotnet.exe is missing. It's time to try msbuild /t:Restore
Also, I wonder if we should keep using paket pack, or switch to msbuild /t:Pack

**Comment by marcinsmialek** [comment]
Why can't this be easy?
Travis CI - MSBuild 16, no .NET CLI
AppVeyor - .NET CLI available, but MSBuild 14

I guess we'll have to make it try .NET CLI if it's present, and if not, then MSBuild 16 if present, and if not, then fail.

**Comment by willsmith9182** [comment]
@marcinsmialek how about just branching based on the build system and using msbuild on travis and .net cli on appveyor?

**Comment by willsmith9182** [comment]
> Also, I wonder if we should keep using paket pack, or switch to msbuild /t:Pack

which ever is easier I guess? I have no strong feelings

> I guess we'll have to make it try .NET CLI if it's present, and if not, then MSBuild 16 if present, and if not, then fail.

sounds like a good idea

File: build.sh [hunk 1]
```diff
@@ -80,8 +80,11 @@ if [ ! -f "$PAKET_EXE" ]; then
     fi
 fi
 
+echo "Found paket.exe in PATH at '$PAKET_EXE'."
```
Comment (willsmith9182) [inline build.sh]: we can remove/change this,

File: build.ps1 [hunk 2]
```diff
@@ -120,8 +120,8 @@ if (!(Test-Path $PAKET_EXE)) {
 Write-Verbose -Message "Found paket.exe in PATH at $PAKET_EXE"
```
Comment (willsmith9182) [inline build.ps1]: we can remove/change this

File: build.sh (hunk 1 above)
Comment (marcinsmialek) [reply build.sh]: Yes, it seems to be too redundant

**Comment by marcinsmialek** [comment]
I upgraded the CI configs to use the latest tooling, and the build configs use .NET CLI, and fall back to MSBuild if it's not found. With the current AppVeyor and Travis CI, it doesn't kick in and .NET CLI works fine.

Locally, I tested all combinations of clean builds, rebuilds, on Windows, Linux, MSBuild, and .NET CLI, and there were no issues. CI seems to be running fine too.

I think it would be good to integrate these changes, so we could work on the "official" code later on. We have some extra options to add, so the current Paket is fully supported. The tests could be expanded too, and code simplified with C# 7 syntax. And we should ensure that everything works fine with .NET Core projects. But I don't think we should do any greater changes before this PR merge.

**Comment by willsmith9182** [comment]
@gep13 are you happy with this?

**Comment by gep13** [comment]
I haven’t had a chance to review, but I will do this tonight. I will defer to you both for most of it though, as Paket is really not something I am familiar with.

**Comment by gep13** [comment]
@willsmith9182 @marcinsmialek would both of you be interested in being added to this Organisation to help with on going maintenance of this addin/module?

**Comment by willsmith9182** [comment]
Yea, I'd hapily maintain this package going forwards. Intending to use paket across our entire repo at werk so it'll benefit me too :)

**Comment by willsmith9182** [comment]
> I haven’t had a chance to review, but I will do this tonight. I will defer to you both for most of it though, as Paket is really not something I am familiar with.

Well, it's building on both CI's now so i'm happy. Will do one last check of the changes 🥇

**Comment by willsmith9182** [comment]
@marcinsmialek  https://github.com/willsmith9182/Cake.Paket/pull/3 adds the nuget project attrs that @gep13 requested

**Comment by gep13** [comment]
@willsmith9182 @marcinsmialek I have added you both as maintainers on this repository.

Do you both use Twitter/Slack?  Would be good to get you both added to the Cake Contrib Slack Team (so that you can ask any questions that you might have) and also to follow your Twitter account from the Cake Contrib account.

Let me know if you have any questions.

**Comment by willsmith9182** [comment]
I don't do twitter but I has some slacks :)

**Comment by gep13** [comment]
@willsmith9182 if you can let me know what email address to use, I can send over an invite to the Slack Team.

**Comment by willsmith9182** [comment]
willsmith182@gmail.com :)

**Comment by gep13** [comment]
@willsmith9182 invite sent.

**Comment by marcinsmialek** [comment]
@gep13 I'm on Twitter with the same handle: @marcinsmialek, and my email is msmialek@gmail.com
There are a few Slack servers I used to join, but not for Cake nor Paket, at least not yet.

I'm one of these people who fix stuff which irritates them, so I'd be glad to be added too :)
I also added both Cake and Paket to the tools used on my current job-project. Improving their maintainability would help.

Thanks

**Comment by gep13** [comment]
@marcinsmialek that is great, thanks!

I have invited you to the Cake Contrib Slack Team.

**Comment by willsmith9182** [comment]
awesome sauce

**Comment by gep13** [comment]
@willsmith9182 did you receive the invitation to join the GitHub Organisation?  I have seen that @marcinsmialek has, and should be able to commit directly to this repository now, but I see that there is still a pending invitation for you.

**Comment by willsmith9182** [comment]
yea all sorted

**Comment by gep13** [comment]
@willsmith9182 glad to hear it! Are you folks in a position to get a new release out, or is there still work to be done before shipping?

**Comment by marcinsmialek** [comment]
A little extra work is needed. I hope I'll be able to finish my tasks tomorrow.

**Pull request merged**